oauth_session.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2012 Jimmy Zelinskie. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package reddit implements an abstraction for the reddit.com API.
+package geddit
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2"
+)
+
+// OAuthSession represents an OAuth session with reddit.com --
+// all authenticated API calls are methods bound to this type.
+type OAuthSession struct {
+	Client       *http.Client
+	ClientID     string
+	ClientSecret string
+	Username     string
+	Password     string
+	OAuthConfig  *oauth2.Config
+}
+
+// NewLoginSession creates a new session for those who want to log into a
+// reddit account via OAuth.
+func NewOAuthSession(username, password, useragent, clientID, clientSecret string) (*OAuthSession, error) {
```

Comment:
I don't necessarily like passing more than a few args to functions (it's hard to remember the order without looking at documentation).

I was thinking maybe we could pass a struct or something instead.

However, I was also thinking we have two constructor functions for OAuthSession types: One for user/password tokens, and another for auth code tokens (asking for a user's permission).